taskcat/_config.py

```diff
@@ -315,8 +316,15 @@ def _create_regional_bucket_obj(self, bucket_objects, region, test):
             name = bucket_objects[_bucket_obj_key].name
             auto_generated = bucket_objects[_bucket_obj_key].auto_generated
         else:
-            name = test.s3_bucket
+            name = f"{test.s3_bucket}-{region.name}"
             auto_generated = False
+            try:
+                region.client("s3").head_bucket(Bucket=name)
+            except ClientError as e:
+                if "(404)" in str(e):
```

Comment:
404 isn't the response code. 

```
>>> import boto3
>>> s3 = boto3.client('s3')
>>> s3.head_bucket(Bucket='foobar-doesnt-exist-kklsdjfdklsfj')
Traceback (most recent call last):
  File "<input>", line 1, in <module>
    s3.head_bucket(Bucket='foobar-doesnt-exist-kklsdjfdklsfj')
  File "/usr/local/lib/python3.7/site-packages/botocore/client.py", line 316, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/usr/local/lib/python3.7/site-packages/botocore/client.py", line 626, in _make_api_call
    raise error_class(parsed_response, operation_name)
botocore.exceptions.ClientError: An error occurred (400) when calling the HeadBucket operation: Bad Request


>>> s3.head_bucket(Bucket='aws-quickstart-no-region')
Traceback (most recent call last):
  File "<input>", line 1, in <module>
    s3.head_bucket(Bucket='aws-quickstart-no-region')
  File "/usr/local/lib/python3.7/site-packages/botocore/client.py", line 316, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/usr/local/lib/python3.7/site-packages/botocore/client.py", line 626, in _make_api_call
    raise error_class(parsed_response, operation_name)
botocore.exceptions.ClientError: An error occurred (400) when calling the HeadBucket operation: Bad Request

>>> s3.head_bucket(Bucket='aws-quickstart-us-east-4')
Traceback (most recent call last):
  File "<input>", line 1, in <module>
    s3.head_bucket(Bucket='aws-quickstart-us-east-4')
  File "/usr/local/lib/python3.7/site-packages/botocore/client.py", line 316, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/usr/local/lib/python3.7/site-packages/botocore/client.py", line 626, in _make_api_call
    raise error_class(parsed_response, operation_name)
botocore.exceptions.ClientError: An error occurred (400) when calling the HeadBucket operation: Bad Request
>>>
```

Reply:
Interesting, the docs indicate this may return responses such as 403 or 404. 

It also indicates it'll return a `NoSuchBucket` exception.

https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3.html#S3.Client.head_bucket

```
>>> import boto3
>>> s3 = boto3.client('s3')
>>> s3.head_bucket(Bucket='aws-quickstart-us-east-4')
Traceback (most recent call last):
  File "<input>", line 1, in <module>
    s3.head_bucket(Bucket='aws-quickstart-us-east-4')
  File "/Users/andglenn/.virtualenvs/boto3_1136/lib/python3.7/site-packages/botocore/client.py", line 316, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/Users/andglenn/.virtualenvs/boto3_1136/lib/python3.7/site-packages/botocore/client.py", line 626, in _make_api_call
    raise error_class(parsed_response, operation_name)
botocore.exceptions.ClientError: An error occurred (400) when calling the HeadBucket operation: Bad Request
>>> boto3.__version__
'1.13.6'
>>>
```

Reply:
I don't get the same as you

```
>>> import boto3
>>> boto3.__version__
'1.12.21'

>>> boto3.client("s3", region_name="us-east-1").head_bucket(Bucket="test") # exists, but no access
Traceback (most recent call last):
  File "<input>", line 1, in <module>
  File "/Users/jmmccon/workspace/taskcat/.venv/lib/python3.8/site-packages/botocore/client.py", line 316, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/Users/jmmccon/workspace/taskcat/.venv/lib/python3.8/site-packages/botocore/client.py", line 626, in _make_api_call
    raise error_class(parsed_response, operation_name)
botocore.exceptions.ClientError: An error occurred (403) when calling the HeadBucket operation: Forbidden

>>> boto3.client("s3", region_name="us-east-1").head_bucket(Bucket="test-doesnt-exist") # doesn;t exist
Traceback (most recent call last):
  File "<input>", line 1, in <module>
  File "/Users/jmmccon/workspace/taskcat/.venv/lib/python3.8/site-packages/botocore/client.py", line 316, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/Users/jmmccon/workspace/taskcat/.venv/lib/python3.8/site-packages/botocore/client.py", line 626, in _make_api_call
    raise error_class(parsed_response, operation_name)
botocore.exceptions.ClientError: An error occurred (404) when calling the HeadBucket operation: Not Found
```